client/my-sites/email/email-providers-stacked-comparison/email-provider-stacked-card/email-provider-generic-form.tsx

```diff
@@ -0,0 +1,868 @@
+	autoFocus?: boolean;
+	children?: ReactNode;
+	domains?: any;
+	extraValidation: ( user: GenericNewUser ) => GenericNewUser;
+	selectedDomainName: string;
+	setValidForm: ( valid: boolean ) => void;
+	showAddAnotherMailboxButton: boolean;
+	onUsersChange: ( users: GenericNewUser[] ) => void;
+	onReturnKeyPress: ( event: Event ) => void;
+	hiddenFields: string[];
+	users: GenericNewUser[];
+}
+
+export const EmailProviderGenericForm: FunctionComponent< EmailProviderGenericFormProps > = ( {
+	autoFocus = false,
+	children,
+	domains,
+	extraValidation,
+	onUsersChange,
+	onReturnKeyPress,
+	hiddenFields = [],
+	selectedDomainName,
+	setValidForm,
+	showAddAnotherMailboxButton = true,
+	users,
+} ) => {
+	const translate = useTranslate();
+
+	const onUserValueChange = ( uuid: string ) => (
+		fieldName: string,
+		fieldValue: string,
+		mailBoxFieldTouched = false
+	) => {
+		const changedUsers = users.map( ( user ) => {
+			if ( user.uuid !== uuid ) {
+				return user;
+			}
+
+			const changedUser = { ...user, [ fieldName ]: { value: fieldValue, error: null } };
+
+			if ( GENERIC_EMAIL_FORM_FIRST_NAME_FIELD === fieldName && ! mailBoxFieldTouched ) {
+				return { ...changedUser, mailBox: { value: sanitizeEmail( fieldValue ), error: null } };
+			}
+
+			return changedUser;
+		} );
+
+		onUsersChange( validateUsers( changedUsers, extraValidation, hiddenFields ) );
+		setValidForm( areAllUsersValid( users, hiddenFields ) );
+	};
+
+	const onUserAdd = () => {
+		onUsersChange( [ ...users, newUser( selectedDomainName ) ] );
+	};
+
+	const onUserRemove = ( uuid: string ) => () => {
+		const newUserList = users.filter( ( _user ) => _user.uuid !== uuid );
+
+		onUsersChange( 0 < newUserList.length ? newUserList : [ newUser( selectedDomainName ) ] );
+	};
+
+	return (
+		<div>
+			{ users.map( ( user, index ) => (
+				<Fragment key={ user.uuid }>
+					<GenericNewUser
+						autoFocus={ autoFocus } // eslint-disable-line jsx-a11y/no-autofocus
+						domains={
+							domains ? domains.map( ( domain: any ) => domain.name ) : [ selectedDomainName ]
+						}
+						hiddenFields={ hiddenFields }
+						user={ user }
+						onUserValueChange={ onUserValueChange( user.uuid ) }
+						onUserRemove={ onUserRemove( user.uuid ) }
+						onReturnKeyPress={ onReturnKeyPress }
+						selectedDomainName={ selectedDomainName }
+						showTrashButton={ index > 0 }
+					/>
+
+					<hr className="email-provider-generic-form__user-divider" />
+				</Fragment>
+			) ) }
+
+			{ showAddAnotherMailboxButton && (
+				<div className="email-provider-generic-form__add-another-user-container">
+					{ /* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/interactive-supports-focus */ }
+					<span
+						role="button"
+						className="email-provider-generic-form__add-another-user-button"
+						onClick={ onUserAdd }
+					>
+						<Gridicon icon="plus" />
+						{ translate( 'Add an extra mailbox' ) }
+					</span>
```

Comment:
Can you use the `Button` element with the `borderless` attribute set instead of a `span`? The `jsx-a11y` lint disables look really dangerous.